src/Core/Framework/Plugin/Command/PluginZipImportCommand.php

```diff
@@ -0,0 +1,78 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Plugin\Command;
+
+use Composer\IO\ConsoleIO;
+use Shopware\Core\Framework\Adapter\Console\ShopwareStyle;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Plugin\Exception\NoPluginFoundInZipException;
+use Shopware\Core\Framework\Plugin\PluginManagementService;
+use Shopware\Core\Framework\Plugin\PluginService;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\HelperSet;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class PluginZipImportCommand extends Command
+{
+    protected static $defaultName = 'plugin:zip-import';
+
+    /**
+     * @var PluginManagementService
+     */
+    private $pluginManagementService;
+
+    /**
+     * @var PluginService
+     */
+    private $pluginService;
+
+    public function __construct(PluginManagementService $pluginManagementService, PluginService $pluginService)
+    {
+        parent::__construct();
+        $this->pluginManagementService = $pluginManagementService;
+        $this->pluginService = $pluginService;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure(): void
+    {
+        $this->setDescription('Import plugin zip file.')
+            ->addArgument('zip-file', InputArgument::REQUIRED, 'Zip file that contains a shopware platform plugin.')
+            ->addOption('no-refresh', null, InputOption::VALUE_OPTIONAL, 'Do not refresh plugin list.');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $zipFile = $input->getArgument('zip-file');
+        $io = new ShopwareStyle($input, $output);
+        $io->title('Shopware Plugin Zip Import');
+
+        try {
+            $this->pluginManagementService->extractPluginZip($zipFile, false);
```

Comment:
Shouldn't there be an option to delete the zip?

Reply:
In my case it is not intended to delete it as I want to do the more than once in a short time. I can add an option but I prefer it to be non-deleting as default.

Reply:
that would be ok 👍